### PR TITLE
Fixing builds and standardize on a config file name

### PR DESCRIPTION
**Pull request by shawn-hurley**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@ build:
 install:
 	cp broker ${PREFIX}/bin/ansible-service-broker
```
Comment (shawn-hurley) [inline Makefile]: Should we rename to asbd @jmontleon @jwmatthews

Comment (jmontleon) [inline Makefile]: +1 to changing it here and making it match.

File: README.md [hunk 2]
```diff
@@ -62,11 +62,11 @@ make vendor
 ```
 
 **Config**
-A broker is configured via the `broker-config.yaml` file. It's recommended to
-copy over `etc/example-broker-config.yaml` to `etc/broker-config.yaml`, and edit
+A broker is configured via the `config.yaml` file. It's recommended to
+copy over `etc/example-config.yaml` to `etc/config.yaml`, and edit
 as desired.
```
Comment (jmontleon) [inline README.md]: "to `/etc/ansible-service-broker/config.yaml`"

File: README.md [hunk 3]
```diff
@@ -97,7 +97,7 @@ configurations.
 ### Misc Targets
 * `make clean`: Delete binaries built from source
 =======
-* `make run`: Runs the broker with the default profile, configured via `/etc/broker-config.yaml`
+* `make run`: Runs the broker with the default profile, configured via `/etc/config.yaml`
```
Comment (jmontleon) [inline README.md]: I'm not sure what makes sense here:
I see in the new prep and run scripts:
`BROKER_CONFIG=${GENERATED_BROKER_CONFIG}` and
`GENERATED_BROKER_CONFIG="../etc/generated_local_development.yaml"`

File: Makefile (hunk 1 above)
Comment (jwmatthews) [inline Makefile]: Unsure, I don't understand what we use "install" for.

File: README.md [hunk 4]
```diff
@@ -97,7 +97,7 @@ configurations.
 ### Misc Targets
 * `make clean`: Delete binaries built from source
 =======
```
Comment (shawn-hurley) [inline README.md]: Fixing Merge conflict lines here

**Comment by jmrodri** [comment]
## TEST make install
```
[jesusr@speed3 ansible-service-broker{shawn-hurley-config-docker-change}]$ sudo make install
cp broker /usr/local/bin/ansible-service-broker
mkdir -p /usr/local/etc/ansible-service-broker
cp etc/example-config.yaml /usr/local/etc/ansible-service-broker/config.yaml

[jesusr@speed3 ansible-service-broker{shawn-hurley-config-docker-change}]$ /usr/local/bin/ansible-service-broker 
============================================================
==           Starting Ansible Service Broker...           ==
============================================================
ERROR: Failed to read config file
stat /etc/ansible-service-broker/config.yaml: no such file or directory
[jesusr@speed3 ansible-service-broker{shawn-hurley-config-docker-change}]$ 

```
The config gets installed to the wrong place using the default PREFIX or is this correct?

**Comment by jmrodri** [comment]
### TEST make install PREFIX=/
```
[jesusr@speed3 ansible-service-broker{shawn-hurley-config-docker-change}]$ sudo make install PREFIX=/
cp broker //bin/ansible-service-broker
mkdir -p //etc/ansible-service-broker
cp etc/example-config.yaml //etc/ansible-service-broker/config.yaml
[jesusr@speed3 ansible-service-broker{shawn-hurley-config-docker-change}]$ /usr/bin/ansible-service-broker 
============================================================
==           Starting Ansible Service Broker...           ==
============================================================
ERROR: Failed to initialize logger
open /var/log/ansible-service-broker/asb.log: no such file or directory
```

install doesn't create all the required directories. I think this was broken for sometime.

File: README.md (hunk 3 above)
Comment (jwmatthews) [inline README.md]: I'd change this to:

 `make run`: Runs the broker with a generated configuration file

File: ansible-service-broker.spec [hunk 5]
```diff
@@ -191,12 +191,9 @@ rm -rf src
 install -d -p %{buildroot}%{_bindir}
 install -p -m 755 broker %{buildroot}%{_bindir}/asbd
 install -p -m 755 build/entrypoint.sh %{buildroot}%{_bindir}/entrypoint.sh
-install -p -m 755 build/%{name} %{buildroot}%{_bindir}/%{name}
 sed -i 's,/usr/local/%{name}/bin,/usr/libexec/%{name},g' %{buildroot}%{_bindir}/%{name}
-install -d -p %{buildroot}%{_docdir}/%{name}
 install -d -p %{buildroot}%{_sysconfdir}/%{name}
-install -p -m 755 etc/example-broker-config.yaml %{buildroot}%{_docdir}/%{name}/example-broker-config.yaml
-install -p -m 644 build/broker-config.yaml %{buildroot}%{_sysconfdir}/%{name}/broker-config.yaml
+install -p -m 644 etc/example-config.yaml %{buildroot}%{_sysconfdir}/%{name}/config.yaml
```
Comment (jwmatthews) [inline ansible-service-broker.spec]: @jmontleon do we want the example config file to be:
 etc/config.yaml.example in our source?

So a rename of that file is required?

**Comment by jmrodri** [comment]
### TEST tito build --test --rpm
```
[jesusr@speed3 ansible-service-broker{shawn-hurley-config-docker-change}]$ tito build --test --rpm
Building package [ansible-service-broker-0.9.2-1]
Wrote: /tmp/tito/ansible-service-broker-git-2.57dc9f7.tar.gz
gobuild: invalid option -- 't'
error: Unknown option t in gobuild(o:)

Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.Ij9KEE

LOTS OF OUTPUT

Executing(%install): /bin/sh -e /var/tmp/rpm-tmp.xKEKMd
+ umask 022
+ cd /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILD
+ '[' /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILDROOT/ansible-service-broker-0.9.2-1.git.2.57dc9f7.fc25.x86_64 '!=' / ']'
+ rm -rf /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILDROOT/ansible-service-broker-0.9.2-1.git.2.57dc9f7.fc25.x86_64
++ dirname /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILDROOT/ansible-service-broker-0.9.2-1.git.2.57dc9f7.fc25.x86_64
+ mkdir -p /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILDROOT
+ mkdir /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILDROOT/ansible-service-broker-0.9.2-1.git.2.57dc9f7.fc25.x86_64
+ cd ansible-service-broker-git-2.57dc9f7
+ install -d -p /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILDROOT/ansible-service-broker-0.9.2-1.git.2.57dc9f7.fc25.x86_64/usr/bin
+ install -p -m 755 broker /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILDROOT/ansible-service-broker-0.9.2-1.git.2.57dc9f7.fc25.x86_64/usr/bin/asbd
+ install -p -m 755 build/entrypoint.sh /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILDROOT/ansible-service-broker-0.9.2-1.git.2.57dc9f7.fc25.x86_64/usr/bin/entrypoint.sh
+ sed -i s,/usr/local/ansible-service-broker/bin,/usr/libexec/ansible-service-broker,g /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILDROOT/ansible-service-broker-0.9.2-1.git.2.57dc9f7.fc25.x86_64/usr/bin/ansible-service-broker
sed: can't read /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILDROOT/ansible-service-broker-0.9.2-1.git.2.57dc9f7.fc25.x86_64/usr/bin/ansible-service-broker: No such file or directory
error: Bad exit status from /var/tmp/rpm-tmp.xKEKMd (%install)


RPM build errors:
    Unknown option t in gobuild(o:)
    Bad exit status from /var/tmp/rpm-tmp.xKEKMd (%install)


ERROR: Error running command: rpmbuild --define "_source_filedigest_algorithm md5" --define "_binary_filedigest_algorithm md5"   --eval '%undefine scl' --define "_topdir /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3" --define "_sourcedir /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/SOURCES" --define "_builddir /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/BUILD" --define "_srcrpmdir /tmp/tito" --define "_rpmdir /tmp/tito"   --clean  -ba /tmp/tito/rpmbuild-ansible-service-broker5nl3y1r3/SOURCES/ansible-service-broker-git-2.57dc9f7/ansible-service-broker.spec

```

The sed call is failing because we removed a line.

**Comment by jmrodri** [comment]
### TEST tito build --test --rpm PASS

```
[jesusr@speed3 ansible-service-broker{shawn-hurley-config-docker-change}]$ rpm -qlp /tmp/tito/x86_64/ansible-service-broker-0.9.2-1.git.3.b7b1242.fc25.x86_64.rpm
/etc/ansible-service-broker
/etc/ansible-service-broker/config.yaml
/usr/bin/asbd
/usr/lib/systemd/system/ansible-service-broker.service
/usr/libexec/ansible-service-broker
/usr/libexec/ansible-service-broker/asbcli
/usr/libexec/ansible-service-broker/asbcli/README.md
/usr/libexec/ansible-service-broker/asbcli/asbcli
/usr/libexec/ansible-service-broker/asbcli/requirements.txt
/usr/libexec/ansible-service-broker/deploy.sh
/usr/libexec/ansible-service-broker/my_local_dev_vars.example
/usr/libexec/ansible-service-broker/prep_local_devel_env.sh
/usr/libexec/ansible-service-broker/run_local.sh
/usr/libexec/ansible-service-broker/test_local_server_connection.sh
/usr/libexec/ansible-service-broker/travis.sh
/usr/share/licenses/ansible-service-broker
/usr/share/licenses/ansible-service-broker/LICENSE
/var/log/ansible-service-broker
/var/log/ansible-service-broker/asb.log
```

**Pull request merged**